### PR TITLE
Fix Mypy errors in routing and testclient

### PR DESCRIPTION
**Pull request by aminalaee**
Mypy 0.920 was released last night and causes failures in routing and test client.

https://mypy-lang.blogspot.com/2021/12/mypy-0920-released.html

### CODE REVIEW

File: starlette/testclient.py [hunk 1]
```diff
@@ -151,6 +151,8 @@ def send(
             for key, value in request.headers.items()
         ]
 
+        scope: typing.Dict[str, typing.Any] = {}
```
Comment (Kludex) [inline starlette/testclient.py]: Does this work?
```suggestion
        scope: typing.Dict[str, typing.Any]
```

Comment (aminalaee) [reply starlette/testclient.py]: yup

**Review by Kludex — APPROVED** [review]
Thanks :)

**Comment by Kludex** [comment]
We should still pin the dependencies and enable dependabot, like the other encode projects. (just as a note)

**Pull request merged**